x/auth/types/account.go

```diff
@@ -169,6 +171,16 @@ func (acc BaseAccount) MarshalYAML() (interface{}, error) {
 	return string(bs), err
 }
 
+// Validate checks for errors on the account fields
+func (acc BaseAccount) Validate() error {
```

Comment:
We should not introduce a breaking change here and require the pubkey. In addition, where is the actual implementation of the `GenesisAccount`? The current [`GenesisAccount`](https://github.com/cosmos/cosmos-sdk/blob/master/x/genaccounts/internal/types/genesis_account.go#L29) has module account fields.

Reply:
>We should not introduce a breaking change here and require the pubkey.

it doesn't require the pubkey. If it's defined it checks that the address is derived from it. 

> I noticed various types implement the GenesisAccount interface. But there needs to be an actual implementation of the account. Maybe this lives in Gaia?

why do we need to implement the type? shouldn't the interface suffice?

Reply:
The type needs to be defined in the application.